src/world/area_kmr/kmr_12/events.c

```diff
@@ -0,0 +1,221 @@
+    SI_END(),
+};
+
+StaticNpcSettings goomba_npc_settings = {
+    .height = 20,
+    .radius = 23,
+    .aiScript = &goomba_ai,
+    .hitScript = (Bytecode*)0x80077F70,
+    .defeatScript = (Bytecode*)0x8007809C,
+    .level = 5,
+};
+
+// *INDENT-OFF*
+Script read_west_sign = {
+    SI_GROUP(0),
+
+    // "Eat a Mushroom to regain your energy!"
+    SI_SUSPEND_GROUP(1),
+    SI_CALL(DisablePlayerInput, TRUE),
+    SI_CALL(ShowMessageAtScreenPos, 0x1D0167, 160, 40),
+    SI_RESUME_GROUP(1),
+
+    SI_SET(SI_FLAG(0), FALSE),
+    SI_CALL(get_goomba_ref),
+    SI_IF_NE(SI_VAR(0), FALSE),
+        SI_CALL(GetNpcVar, NpcId_GOOMBA, 0, SI_VAR(0)),
+        SI_IF_EQ(SI_VAR(0), FALSE),
+            // Trigger Goomba to peel off
+            SI_CALL(SetNpcVar, NpcId_GOOMBA, 0, TRUE),
+            SI_SET(SI_FLAG(0), TRUE),
+            SI_WAIT_FRAMES(10),
+        SI_END_IF(),
+    SI_END_IF(),
+    SI_CALL(DisablePlayerInput, FALSE),
+    SI_IF_EQ(SI_FLAG(0), TRUE),
+        SI_UNBIND_ME(),
+    SI_END_IF(),
+
+    SI_END(),
+    SI_RETURN(), // Whoops!
+};
+
+Script goomba_idle = {
+    SI_WAIT_FRAMES(1),
+
+    SI_CALL(SetSelfVar, 0, FALSE),
+    SI_CALL(SetNpcAnimation, NpcId_SELF, ANIMATION(SpriteId_GOOMBA, 0, 13)),
+    SI_CALL(EnableNpcShadow, NpcId_SELF, FALSE),
+    SI_CALL(SetSelfEnemyFlagBits, 0x00000020, TRUE),
+
+    // Wait until read_sign sets NPC var 0
+    SI_LABEL(0),
+    SI_CALL(GetSelfVar, 0, SI_VAR(0)),
+    SI_WAIT_FRAMES(1),
+    SI_IF_EQ(SI_VAR(0), FALSE),
+        SI_GOTO(0),
+    SI_END_IF(),
+
+    // Peel and jump off the sign
+    SI_CALL(SetNpcFlagBits, NpcId_SELF, 0x00240000, TRUE),
+    SI_WAIT_FRAMES(3),
+    SI_SET_F(SI_VAR(0), SI_FIXED(0.0f)),
+    SI_LOOP(9),
+        SI_ADD_F(SI_VAR(0), SI_FIXED(10.0f)),
+        SI_CALL(SetNpcRotation, NpcId_SELF, 0, SI_VAR(0), 0),
+        SI_WAIT_FRAMES(1),
+    SI_END_LOOP(),
+    SI_CALL(SetNpcAnimation, NpcId_SELF, ANIMATION(SpriteId_GOOMBA, 0, 0)),
+    SI_LOOP(9),
+        SI_ADD_F(SI_VAR(0), SI_FIXED(10.0f)),
+        SI_CALL(SetNpcRotation, NpcId_SELF, 0, SI_VAR(0), 0),
+        SI_WAIT_FRAMES(1),
+    SI_END_LOOP(),
+    SI_CALL(SetNpcAnimation, NpcId_SELF, ANIMATION(SpriteId_GOOMBA, 0, 7)),
+    SI_WAIT_FRAMES(20),
+    SI_CALL(SetNpcAnimation, NpcId_SELF, ANIMATION(SpriteId_GOOMBA, 0, 1)),
+    SI_CALL(PlaySoundAtNpc, NpcId_SELF, 248, 0),
+    SI_CALL(func_802CFE2C, NpcId_SELF, 8192),
+    SI_CALL(func_802CFD30, NpcId_SELF, 5, 6, 1, 1, 0),
+    SI_WAIT_FRAMES(12),
+    SI_WAIT_FRAMES(5),
+    SI_CALL(PlaySoundAtNpc, NpcId_SELF, 812, 0),
+    SI_CALL(EnableNpcShadow, NpcId_SELF, TRUE),
+    SI_CALL(SetNpcJumpscale, NpcId_SELF, SI_FIXED(0.6005859375f)),
+    SI_CALL(NpcJump0, NpcId_SELF, -35, 0, 30, 23),
+    SI_CALL(func_802CFD30, NpcId_SELF, 0, 0, 0, 0, 0),
+    SI_CALL(InterpNpcYaw, NpcId_SELF, 90, 0),
+    SI_CALL(SetNpcFlagBits, NpcId_SELF, 0x00240000, FALSE),
+    SI_CALL(SetSelfEnemyFlagBits, 0x00000020, FALSE),
+    SI_CALL(SetSelfEnemyFlagBits, 0x40000000, TRUE),
```

Comment:
With these flags being used so often, it would probably be worth making defines and using those, even if they are just _UNK_X etc

Reply:
@JoshDuMan said they would make a future PR that does this for all flags in the source, so I'll consider this out-of-scope for my PR.